src/pkg/ingress/bindings/binding_config.go

```diff
@@ -43,12 +43,39 @@ func (d *DrainParamParser) FetchBindings() ([]syslog.Binding, error) {
 			b.InternalTls = true
 		}
 
+		b.Type = getBindingType(urlParsed)
+
 		processed = append(processed, b)
 	}
 
 	return processed, nil
 }
 
+func getBindingType(u *url.URL) syslog.BindingType {
+	bindingType := syslog.BINDING_TYPE_LOG
+	switch u.Query().Get("drain-type") {
+	case "metrics":
+		bindingType = syslog.BINDING_TYPE_METRIC
+	case "all":
+		bindingType = syslog.BINDING_TYPE_ALL
+	case "allWithTimers":
+		bindingType = syslog.BINDING_TYPE_AGGREGATE
+	}
+
+	if u.Query().Get("include-metrics-deprecated") != "" {
+		bindingType = syslog.BINDING_TYPE_AGGREGATE
+	}
+	return bindingType
+}
+
+func getRemoveMetadataQuery(u *url.URL) string {
```

Comment:
This is cool way to avoid breaking changes in the API and keep backward compatibility. Well done! :)

Reply:
I understand avoiding breaking changes is important :D